metrics/web/docs/features/deep_links/01_deep_linking.md

```diff
@@ -0,0 +1,106 @@
+
+# Feasibility study
+> A preliminary study of the feasibility of implementing this feature.
+
+The `Deep Linking` feature improves the overall user experience in terms of navigation, so this feature makes sense for the project.
+
+To be able to implement the `Deep Linking` feature we should be able to perform the following actions:
+- Parsing the `query parameters` from the URL;
+- Updating the URL and the `query parameters` in response to any events.
+
+Since `Navigator 2.0` provides the navigation system that allows performing the actions listed above, and the Metrics Web application uses `Navigator 2.0`, it is possible to implement the `Deep Linking` feature in the Metrics Web application.
+
+# Landscape
+> Look for existing solutions in the area.
+
+There are 2 main approaches for implementing the `Deep Linking` feature in the Metrics Web application:
+- Using 3-rd party package (such as [`routemaster`](https://pub.dev/packages/routemaster), [`beamer`](https://pub.dev/packages/beamer), etc.);
+- A custom implementation based on the existing `Navigator 2.0` integration.
+
+Let's compare the listed approaches in a bit more detail.
+
+## 3-rd party package approach
+Consider the following pros and cons of the 3-rd party package utilization approach:
+
+Pros:
+- Provides ready-to-use implementation out of the box, including deep linking, nested navigation, and many other features.
+
+Cons:
+- May require code generation;
+- Requires the navigation system reintegration - meaning that we must use the 3-rd party `RouteInformationParser`s and `RouterDelegate`s.
+
+## Custom implementation
+Consider the following pros and cons of the custom implementation approach:
+
+Pros:
+- Highly customizable approach;
+- Well-testable approach;
+
+Cons:
+- Requires more code comparing to the 3-rd party package approach.
+
+According to the comparison above, we choose the [`Custom implementation`](#custom-implementation) approach, as it fits the current navigation approach used in the Metrics Web application, and satisfies the customization and testability requirements.
+
+Consider this diagram that briefly describes the custom implementation approach:
+
+![Custom approach diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://raw.githubusercontent.com/Flank/flank-dashboard/update_deep_links_analysis/metrics/web/docs/features/deep_links/diagrams/custom_approach_component_diagram.puml)
+
+# Prototyping
+> Create a simple prototype to confirm that implementing this feature is possible.
+
+The following code snippet demonstrates the process of parsing the query parameters from the URL:
```

Comment:
Would it be great to mention here the things we should be able to perform to implement the deep linking and then take a closer look at each of them? I mean, would it be great to have an opening sentence for this section here or something like that?